### PR TITLE
Amazon Bedrock streaming output

### PR DESCRIPTION
**Pull request by sunbc0120**
Add streaming output for Amazon Bedrock

### CODE REVIEW

File: libs/langchain/langchain/llms/bedrock.py
Comment (baskaryan) [inline libs/langchain/langchain/llms/bedrock.py:39]: cls not self

**Comment by 3coins** [comment]
@sunbc0120 
Thanks for adding this to the Bedrock LLM, would you want to wait for #8017 to be merged before adding this? Also, would you want to add an example of usage in the sample notebook?

**Review by 3coins — COMMENTED** [review]
Doesn't this also need a `streaming` boolean flag and changes to the `_call` to handle streaming?

File: libs/langchain/langchain/llms/bedrock.py
Comment (sunbc0120) [reply libs/langchain/langchain/llms/bedrock.py:39]: Fixed. surprised lint didn't pick this one up.

**Comment by sunbc0120** [comment]
> @sunbc0120 Thanks for adding this to the Bedrock LLM, would you want to wait for #8017 to be merged before adding this? Also, would you want to add an example of usage in the sample notebook?

- example notebook: adding
- wait for #8017 , don't mind wait, but upon checking the changes, maybe it makes sense it updates after this one merged as it says 
 
```python
raise NotImplementedError(
            """Bedrock doesn't support stream requests at the moment."""
        )
```

This won't be valid after this PR merged.

**Comment by sunbc0120** [comment]
> Doesn't this also need a `streaming` boolean flag and changes to the `_call` to handle streaming?

Is there any documentation on this? It was also my 1st thoughts but failed to find any explanation.

**Comment by shabie** [comment]
Not just `_call` for llms but also `_stream` for the chat model class.

File: libs/langchain/langchain/llms/bedrock.py
Comment (nfcampos) [inline libs/langchain/langchain/llms/bedrock.py:50]: If a new provider is introduced with a different output key we don't expect here, is it better to throw a more meaningful error instead of leaving this as a future cryptic KeyError?

File: libs/langchain/langchain/llms/bedrock.py
Comment (nfcampos) [inline libs/langchain/langchain/llms/bedrock.py:193]: Looks like we're not enforcing the stop tokens in `_stream`, unlike in `_call`, see line 234. What we probably want to do is something like
```py
buffer = ''
for event in stream:
  text = ...
  buffer += text
  if enforce_stop_tokens(buffer, stop).endswith(text):
    yield GenerationChunk(text=text)
  else:
    break

**Review by nfcampos — CHANGES_REQUESTED** [review]

File: libs/langchain/langchain/llms/bedrock.py
Comment (sansmoraxz) [reply libs/langchain/langchain/llms/bedrock.py:50]: This is for `amazon.titan` streaming output if I am not wrong. Also last I checked AI21's Jurassic models don't support streaming last I checked. Ideally this should throw an error.

**Comment by sunbc0120** [comment]
Working on with #10393

**Pull request closed without merging**
